packages/ng/date2/date-input/date-input.component.ts

```diff
@@ -246,12 +247,13 @@ export class DateInputComponent extends AbstractDateComponent implements Control
 		}
 	}
 
-	validate(control: AbstractControl<Date, Date>): ValidationErrors {
+	validate(control: AbstractControl<Date | string | null>): ValidationErrors | null {
 		// null is not an error but means we'll skip everything else, we'll let the presence of a
 		// Validators.required (or not) decide if it's an error.
 		if (control.value === null || control.value === undefined) {
 			return null;
 		}
+		const date = this.inDateISOFormat() ? transformDateInputToDate(control.value) : (control.value as Date);
```

Comment:
shouldn't we do something like
```suggestion
		const date = typeof control.value === 'string' && this.inDateISOFormat() ? transformDateInputToDate(control.value) : control.value;
```
which would allow to remove the `as Date`?

Reply:
I tried this locally and Typescript still thinks `date` is `string | Date`, meaning that we still have to cast.